api/client/options.proto

```diff
@@ -91,6 +95,7 @@ message CommandLineOptions {
   oneof oneof_request_options {
     // See :option:`--request_options` for details.
     RequestOptions request_options = 12;
+    HeaderSource header_source = 19;
```

Comment:
I think we should bike shed the data model a bit before diving into implementation. My thoughts were that we would ideally structure the record-replay along trace lines, i.e. build up a conceptual trace and then replay it, including various attributes of the request, not just header. This could be streamed, in the way that you plan for headers here, but it would be good to describe the roadmap for the data model here.

Reply:
So right now the `Content-Length:` request header dictates how many bytes NH will upstream for a request in the entity body, so it's not explicitly modeled in the proto.. I pinged you on Slack to bikeshed about that :-)

Reply:
Yeah, I reckon it makes sense to create the trace format that we want to be heading for, even if we just replicate constants in it today.